tests/test_trial.py

```diff
@@ -455,6 +454,11 @@ def test_suggest_int_log(storage_init_func):
             warnings.simplefilter("ignore", category=UserWarning)
             trial.suggest_int("z", 0.5, 10, log=True)  # type: ignore
 
+    with pytest.raises(ValueError):
+        study = create_study(storage_init_func(), sampler=sampler)
+        trial = Trial(study, study._storage.create_new_trial(study._study_id))
+        trial.suggest_int("w", 1, 3, step=2, log=True)
```

Comment:
[nit] How about extracting the setup of `study` and `trial` to clarify the error is raised in `trial.suggest_int`?
```suggestion
    study = create_study(storage_init_func(), sampler=sampler)
    trial = Trial(study, study._storage.create_new_trial(study._study_id))
    with pytest.raises(ValueError):
        trial.suggest_int("w", 1, 3, step=2, log=True)
```